core/vm/zk_counters.go

```diff
@@ -1883,3 +1883,61 @@ func getRelevantNumberBytes(input []byte) int {
 	}
 	return totalLength - rel
 }
+
+func (cc *CounterCollector) preP256Verify(r, s, x, y *big.Int) {
+	cc.Deduct(S, 50)
+	cc.Deduct(B, 1)
+	cc.multiCall(cc.readFromCallDataOffset, 5)
+	cc.p256verify(r, s, x, y)
+	cc.mStore32()
+	cc.mStoreX()
+
+}
+
+var (
+	SECP256R1_N           = uint256.MustFromHex("0xffffffff00000000ffffffffffffffffbce6faada7179e84f3b9cac2fc632551").ToBig()
+	SECP256R1_N_MINUS_ONE = SECP256R1_N.Sub(SECP256R1_N, big.NewInt(1))
+	SECP256R1_P           = uint256.MustFromHex("0xffffffff00000001000000000000000000000000ffffffffffffffffffffffff").ToBig()
+	SECP256R1_P_MINUS_ONE = SECP256R1_N.Sub(SECP256R1_P, big.NewInt(1))
+	SECP256R1_A           = uint256.MustFromHex("0xffffffff00000001000000000000000000000000fffffffffffffffffffffffc").ToBig()
+	SECP256R1_B           = uint256.MustFromHex("0x5ac635d8aa3a93e7b3ebbd55769886bc651d06b0cc53b0f63bce3c3e27d2604b").ToBig()
+)
+
+func (cc *CounterCollector) p256verify(r, s, x, y *big.Int) {
+	if r.Cmp(big.NewInt(0)) == 0 {
+		cc.Deduct(S, 13)
+		cc.Deduct(B, 1)
+	} else if SECP256R1_N_MINUS_ONE.Cmp(r) == -1 {
+		cc.Deduct(S, 15)
+		cc.Deduct(B, 2)
+	} else if s.Cmp(big.NewInt(0)) == 0 {
+		cc.Deduct(S, 17)
+		cc.Deduct(B, 3)
+	} else if SECP256R1_N_MINUS_ONE.Cmp(s) == -1 {
+		cc.Deduct(S, 19)
+		cc.Deduct(B, 4)
+	} else if SECP256R1_P_MINUS_ONE.Cmp(x) == -1 {
+		cc.Deduct(S, 22)
+		cc.Deduct(B, 5)
+	} else if SECP256R1_P_MINUS_ONE.Cmp(y) == -1 {
+		cc.Deduct(S, 24)
+		cc.Deduct(B, 6)
+	} else if x.Cmp(big.NewInt(0)) == 0 && y.Cmp(big.NewInt(0)) == 0 {
+		cc.Deduct(S, 29)
+		cc.Deduct(B, 8)
+	} else {
+		aux_x3 := x.Exp(x, big.NewInt(3), SECP256R1_P)
```

Comment:
calling `Exp` directly on `x` will mutate the value of x.  We might not want that side effect so it will be safer to use a "new" big int for these values.